### PR TITLE
refactor(EditUpdateCards): add typescript types

### PR DESCRIPTION
**Pull request by szinta**
Contributes to #4286 

Add typescript types to EditUpdateCards component

#### What did you change?
packages/ibm-products/src/components/EditUpdateCards/EditUpdateCards.tsx

#### How did you test and verify your work?
Storybook

### CODE REVIEW

File: packages/ibm-products/src/components/EditUpdateCards/EditUpdateCards.tsx [hunk 1]
```diff
@@ -35,6 +37,102 @@ const componentName = 'EditUpdateCards';
...
+  onClick?(): void;
+  iconDescription: string;
+  href?: string;
+};
+
+interface EditUpdateCardsProps {
+  /**
+   * Icons that are displayed on card. Refer to design documentation for implementation guidelines
+   */
+  actionIcons: Array<ActionIcon>;
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/EditUpdateCards/EditUpdateCards.tsx:53]: Should be optional
```suggestion
  actionIcons?: Array<ActionIcon>;
```

File: packages/ibm-products/src/components/EditUpdateCards/EditUpdateCards.tsx [hunk 2]
```diff
@@ -35,6 +37,102 @@ const componentName = 'EditUpdateCards';
...
+   */
+  className?: string;
+  /**
+   * Optional header description
+   */
+  description?: string;
+  /**
+   * Edit mode children
+   */
+  editChildren: ReactNode;
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/EditUpdateCards/EditUpdateCards.tsx:69]: Should be optional
```suggestion
  editChildren?: ReactNode;
```

File: packages/ibm-products/src/components/EditUpdateCards/EditUpdateCards.tsx [hunk 3]
```diff
@@ -35,6 +37,102 @@ const componentName = 'EditUpdateCards';
...
+   */
+  onPrimaryButtonClick?(): void;
+  /**
+   * Function that's called from the secondary button or action icon
+   */
+  onSecondaryButtonClick?(): void;
+  /**
+   * Preview mode children
+   */
+  previewChildren: ReactNode;
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/EditUpdateCards/EditUpdateCards.tsx:89]: Should be optional
```suggestion
  previewChildren?: ReactNode;
```

**Review by matthewgallo — CHANGES_REQUESTED** [review]
Just found a few props that should be updated as optional as specified in the prop types, other than that looks great.

**Comment by szinta** [comment]
Thanks @matthewgallo for the review. I've applied the suggested changes. Please take a look again.

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**